lib/mondrian/olap/cube.rb

```diff
@@ -77,6 +78,16 @@ def member_by_segments(*segment_names)
         raw_member = @raw_cube.lookupMember(segment_list)
         raw_member && Member.new(raw_member)
       end
+
+      def flush_region_cache_with_segments(*segment_names)
+        members = segment_names.map { |name| member_by_segments(*name).mondrian_member }
+        @cache_control.flush(members)
+      end
+
+      def flush_region_cache_with_full_names(*full_names)
+        members = full_names.map { |name| member(*name).mondrian_member }
+        @cache_control.flush(members)
+      end
```

Comment:
Better delegate `flush_region_cache_with_segments` and `flush_region_cache_with_full_names` to `cache_control` as then all cache specific functionality will be in `CacheControl` class.